compiler/luci-interpreter/src/kernels/Add.test.cpp

```diff
@@ -27,57 +27,106 @@ namespace
 
 using namespace testing;
 
-TEST(AddTest, Float)
+// for quantized Add, the error shouldn't exceed step
+float GetTolerance(float min, float max)
 {
-  Shape base_shape = {2, 3, 1, 2};
-  std::vector<Shape> test_shapes{{1, 1, 3, 2}, {1, 3, 1, 2}, {2, 1, 3, 1}, {2, 3, 1, 1}};
-  std::vector<std::vector<float>> test_outputs = {
-      {0.0f, 2.6f, 0.0f, 2.8f, 0.7f, 3.2f, 1.1f, 0.8f, 0.5f, 1.0f, 1.9f, 1.4f,
-       1.0f, 0.0f, 0.4f, 0.0f, 1.8f, 0.0f, 1.4f, 3.1f, 0.8f, 3.3f, 2.2f, 3.7f,
-       0.0f, 0.3f, 0.0f, 0.5f, 0.0f, 0.9f, 0.9f, 0.0f, 0.3f, 0.0f, 1.7f, 0.0f},
-      {0.0f, 2.6f, 0.5f, 1.0f, 1.8f, 0.0f, 1.4f, 3.1f, 0.0f, 0.5f, 1.7f, 0.0f},
-      {0.0f, 2.5f, 0.0f, 2.6f, 0.0f, 1.9f, 1.1f, 0.7f, 1.2f, 0.8f, 0.5f, 0.1f,
-       1.0f, 0.0f, 1.1f, 0.0f, 0.4f, 0.0f, 1.7f, 3.3f, 2.2f, 3.8f, 2.1f, 3.7f,
-       0.0f, 0.5f, 0.0f, 1.0f, 0.0f, 0.9f, 1.2f, 0.0f, 1.7f, 0.0f, 1.6f, 0.0f},
-      {0.0f, 2.5f, 1.2f, 0.8f, 0.4f, 0.0f, 1.7f, 3.3f, 0.0f, 1.0f, 1.6f, 0.0f}};
-  std::vector<float> input1_data{-0.3f, 2.3f, 0.9f,  0.5f, 0.8f, -1.1f,
-                                 1.2f,  2.8f, -1.6f, 0.0f, 0.7f, -2.2f};
-  std::vector<float> input2_data{0.2f, 0.3f, -0.4f, 0.5f, 1.0f, 0.9f};
-  for (size_t i = 0; i < test_shapes.size(); ++i)
+  float kQuantizedStep = (max - min) / 255.0;
+  return kQuantizedStep;
+}
+
+template <typename T>
+void Check(std::initializer_list<int32_t> input1_shape, std::initializer_list<int32_t> input2_shape,
+           std::initializer_list<int32_t> output_shape, std::initializer_list<float> input1_data,
+           std::initializer_list<float> input2_data, std::initializer_list<float> output_data)
+{
+  float min, max;
+  min =
+      std::min(input1_data) > std::min(input2_data) ? std::min(input2_data) : std::min(input1_data);
+  max =
+      std::max(input1_data) > std::max(input2_data) ? std::max(input1_data) : std::max(input2_data);
+  float kQuantizedTolerance = GetTolerance(min, max);
+  std::pair<float, int32_t> input_quant_param = quantizationParams<T>(min, max);
```

Comment:
Why do input1 and input2 have the same quantization parameter? Why not using separate quantization parameters? ex: input1_quant_param and input2_quant_param

Reply:
Okay, I will update it!